src/Migrator.sol

```diff
@@ -2,167 +2,64 @@
  * @dev BPT: 20WETH-80TOKEN Balancer Pool Token
  * @dev auraBPT: 20WETH-80TOKEN Aura Deposit Vault
  */
-contract Migrator is IMigrator, Ownable {
+contract Migrator is IMigrator {
     using SafeTransferLib for ERC20;
 
-    uint256 public swapSlippage = 10; // 0.1%
-
     uint256 public immutable BPS = 10000;
-    bytes32 public immutable balancerPoolId;
-    WETH public immutable weth;
-    ERC20 public immutable token;
-    ERC20 public immutable balancerPoolToken;
-    ERC20 public immutable auraDepositToken;
-    IUniswapV2Pair public immutable sushiLpToken;
-    IRewardPool4626 public immutable auraPool;
-    AggregatorV3Interface public immutable tokenPrice;
-    IUniswapV2Router02 public immutable sushiRouter;
-    IVault public immutable balancerVault;
-    IBasePool public immutable balancerPool;
-    IAsset public immutable poolAssetWeth;
-    IAsset public immutable poolAssetToken;
-
-    /*
-        Admin functions
-    */
-
-    constructor(InitializationParams memory params) Ownable() {
-        weth = WETH(payable(params.weth));
-        token = ERC20(params.token);
-        balancerPoolToken = ERC20(params.balancerPoolToken);
-        auraDepositToken = ERC20(params.auraDepositToken);
-        sushiLpToken = IUniswapV2Pair(params.sushiLpToken);
-        auraPool = IRewardPool4626(params.auraPool);
-        tokenPrice = AggregatorV3Interface(params.tokenPrice);
-        sushiRouter = IUniswapV2Router02(params.sushiRouter);
-        balancerVault = IVault(params.balancerVault);
-        balancerPool = IBasePool(address(balancerPoolToken));
-        balancerPoolId = balancerPool.getPoolId();
-        poolAssetWeth = IAsset(address(weth));
-        poolAssetToken = IAsset(params.token);
-
-        setApprovals();
-    }
-
-    /// @inheritdoc IMigrator
-    function setSwapSlippage(uint256 _amount) external onlyOwner {
-        require(_amount <= BPS, "swap slippage too high");
-        swapSlippage = _amount;
-    }
-
-    /// @inheritdoc IMigrator
-    function setApprovals() public onlyOwner {
-        ERC20(address(weth)).safeApprove(address(balancerVault), type(uint256).max);
-        token.safeApprove(address(balancerVault), type(uint256).max);
-        balancerPoolToken.safeApprove(address(auraPool), type(uint256).max);
-        ERC20(address(sushiLpToken)).safeApprove(address(sushiRouter), type(uint256).max);
-    }
+    WETH public immutable weth = WETH(payable(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2));
 
     /*
         External functions
     */
 
     /// @inheritdoc IMigrator
-    function migrate(bool _stakeBpt, uint256 _amountTokenMin, uint256 _amountWethMin) external {
-        uint256 slpBalance = sushiLpToken.balanceOf(msg.sender);
-
-        ERC20(address(sushiLpToken)).safeTransferFrom(msg.sender, address(this), slpBalance);
-
-        _unwrapSlp(_amountTokenMin, _amountWethMin);
+    function migrate(MigrationAddresses memory _addresses, MigrationDetails memory _details) external {
```

Comment:
A single struct of params was something I considered but I thought it would be helpful context for someone working on the UI. It also provides a clear separation between the calculations that need to be passed in and the addresses that need to be passed in.

Reply:
consolidated